### PR TITLE
Update RFID endpoint to read from a database

### PR DESCRIPTION
**Pull request by EmberRandall**
This simplifies the existing user/role/ability structure and makes it so the RFID endpoint can read from the database.  It also adds a new endpoint to update users/add new users.

Because even a VM on Windows refuses to run the server (all sorts of weird errors that I haven't been able to fix), this is all tested via the integration tests, but I haven't been able to get it fully up and running.

### CODE REVIEW

**Comment by nallath** [comment]
Hmm, looks like I haven't configured the CI quite right yet, since it doesn't seem to be running them on PR's. 

The tests also seem to be failing locally for me. I will have a look to see if i can get the CI setup correctly.

File: Server/RFIDNamespace.py [hunk 1]
```diff
@@ -17,11 +22,39 @@ class RFID(Resource):
...
             return UNKNOWN_CARD_RESPONSE
         else:
             return "Welcome back!"
 
-
+# name, email are required for new users. Ability can be passed multiple times.
+# Only adds abilities; does not remove them.
+@RFID_namespace.route("/update/<string:card_id>/")
+@RFID_namespace.doc(description ="Update a user's information")
+class RFID(Resource):
```
Comment (nallath) [inline Server/RFIDNamespace.py:35]: MYPY typing is complaining that RFID is defined twice.  

You can see this when running ./checks.sh

File: Server/RFIDNamespace.py [hunk 2]
```diff
@@ -17,11 +22,39 @@ class RFID(Resource):
...
+    @api.response(200, "User updated")
+    @api.response(500, "User update failed")
+    def get(self, card_id):
```
Comment (nallath) [inline Server/RFIDNamespace.py:38]: Doesn't it conceptually make more sense to add new users with a post? As far as I've understood it, a get request shouldn't change any data.

**Review by nallath — CHANGES_REQUESTED** [review]

File: Server/RFIDNamespace.py (hunk 2 above)
Comment (EmberRandall) [reply Server/RFIDNamespace.py:38]: Good call

File: Server/RFIDNamespace.py (hunk 1 above)
Comment (EmberRandall) [reply Server/RFIDNamespace.py:35]: Oops! I wonder why the tests didn't whine about that... fixed now

**Pull request merged**